test-suite/src/showcolumns.rs

```diff
@@ -1,29 +1,77 @@
 use crate::*;
 
+macro_rules! row {
+    ( $( $p:path )* ; $( $v:expr )* ) => (
+        ($( $p($v) ),*)
+    )
+}
+
+macro_rules! showcolumns {
+    ( $( $c: tt )|+ $( ; )? $( $t: path )|+ ; $( $v: expr )+ ; $( $( $v2: expr )+ );+) => ({
+        let mut rows = vec![
+            row!($( $t )+ ; $( $v )+),
+        ];
+
+        gluesql_core::executor::Payload::ShowColumns (
+          //  labels: vec![$( stringify!($c).to_owned().replace("\"", "")),+],
+           concat_with!(rows ; $( $t )+ ; $( $( $v2 )+ );+)
+        )
+    });
+    ( $( $c: tt )|+ $( ; )? $( $t: path )|+ ; $( $v: expr )+ ) => (
+        gluesql_core::executor::Payload::ShowColumns (
+           // labels: vec![$( stringify!($c).to_owned().replace("\"", "")),+],
+            vec![row!($( $t )+ ; $( $v )+ )],
+        )
+    );
+    ( $( $c: tt )|+ $( ; )?) => (
+        gluesql_core::executor::Payload::ShowColumns (
+            //labels: vec![$( stringify!($c).to_owned().replace("\"", "")),+],
+            vec![],
+        )
+    );
+}
+
 test_case!(showcolumns, async move {
+    use gluesql_core::ast::DataType;
     use gluesql_core::{executor::ExecuteError, prelude::Value::*};
 
     run!(
         "
         CREATE TABLE mytable (
+            id8 INT(8),
             id INTEGER,
             rate FLOAT,
+            dec  decimal,
             flag BOOLEAN,
             text TEXT,
-            null_value TEXT NULL,
+            DOB  Date,
+            Tm   Time,
+            ival Interval,
+            tstamp Timestamp,
+            uid    Uuid,
+            hash   Map,
+            glist  List,
         );
     "
     );
 
     test!(
-        Ok(select!(
+        Ok(showcolumns!(
             Field               | Type
             Str                 | Str;
-            "id".to_owned()       "Int".to_owned();
-            "rate".to_owned()     "Float".to_owned();
-            "flag".to_owned()     "Boolean".to_owned();
-            "text".to_owned()     "Text".to_owned();
-            "null_value".to_owned()   "Text".to_owned()
+            "id8".to_owned()      DataType::Int8.to_string();
+            "id".to_owned()       DataType::Int.to_string();
+            "rate".to_owned()     DataType::Float.to_string();
+            "dec".to_owned()      DataType::Decimal.to_string();
+            "flag".to_owned()     DataType::Boolean.to_string();
+            "text".to_owned()     DataType::Text.to_string();
+            "DOB".to_owned()      DataType::Date.to_string();
+            "Tm".to_owned()       DataType::Time.to_string();
+            "ival".to_owned()     DataType::Interval.to_string();
+            "tstamp".to_owned()   DataType::Timestamp.to_string();
+            "uid".to_owned()      DataType::Uuid.to_string();
+            "hash".to_owned()     DataType::Map.to_string();
+            "glist".to_owned()    DataType::List.to_string()
```

Comment:
Oh, I think adding a new macro only for `ShowColumns` is too much.
`select!` simply returns `Payload::Select` and it is for others to read easier than pure `Vec<Vec<Value>>` representation in Rust.
However, in the case of `showcolumns!`, It is simply a type of `Vec<(String, DataType)>` so it would be readable enough to be only with Rust codes without macros.

Reply:
You also don't need to convert `String` and `DataType` to `Value`.
Both `String` and `DataType` has `PartialEq` implementation. We can directly compare with each other
https://github.com/gluesql/gluesql/blob/1213d8288a427745862c0b1c40f83fa057033fd3/core/src/ast/data_type.rs#L3-L5

Reply:
Thanks for the info..  It helped quite a bit..